pkg/apis/elasticsearch/v1alpha1/types.go

```diff
@@ -121,3 +121,44 @@ const (
 	PodStateTypeNotReady PodStateType = "notReady"
 	PodStateTypeFailed   PodStateType = "failed"
 )
+
+// ClusterCondition contains details for the current condition of this elasticsearch cluster.
+type ClusterCondition struct {
+	// Type is the type of the condition.
+	Type ClusterConditionType `json:"type"`
+	// Status is the status of the condition.
+	Status ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,5,opt,name=reason"`
+	// Human-readable message indicating details about last transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,6,opt,name=message"`
+}
+
+// ClusterConditionType is a valid value for ClusterCondition.Type
+type ClusterConditionType string
+
+// These are valid conditions for elasticsearch node
+const (
+	ClusterSettingsUpdateNeeded ClusterConditionType = "SettingsUpdateNeeded"
+	ScalingUp                   ClusterConditionType = "ScalingUp"
+	ScalingDown                 ClusterConditionType = "ScalingDown"
+	Restarting                  ClusterConditionType = "Restarting"
+)
+
+type ConditionStatus string
+
+const (
+	ConditionTrue    ConditionStatus = "True"
+	ConditionFalse   ConditionStatus = "False"
+	ConditionUnknown ConditionStatus = "Unknown"
+)
+
+type ClusterEvent string
```

Comment:
This type seems unused

Reply:
Values of this type mark certain events that happen in the cluster
Eg node quorum is update only if we know know that the cluster scaled-up successfully - `ScaledUp` event.
```go
const (
	ScaledDown            ClusterEvent = "ScaledDown"
	ScaledUp              ClusterEvent = "ScaledUp"
	UpdateClusterSettings ClusterEvent = "UpdateClusterSettings"
	NoEvent               ClusterEvent = "NoEvent"
)
```